### PR TITLE
refactor: Search pipeline run refactoring and baseline unit tests

### PR DESCRIPTION
**Pull request by yuechao-qin**
### TL;DR

Refactored the `list` method in `PipelineRunsApiService_Sql` to improve code organization and added comprehensive test coverage for pipeline run operations.

#### Functional

None.

#### Other

- Extracted filter parsing logic into separate helper functions `_resolve_filter_value` and `_build_filter_where_clauses`
- Moved pipeline run response creation logic into a dedicated `_create_pipeline_run_response` method
- Added constants `_PAGE_TOKEN_OFFSET_KEY` and `_PAGE_TOKEN_FILTER_KEY` for page token field names
- Added comprehensive test suite covering listing, filtering, pagination, annotation CRUD operations, and helper functions

### How to test?

```
uv run pytest tests/test_api_server_sql.py
```

Run the new test suite in `tests/test_api_server_sql.py` which includes:
- Tests for listing pipeline runs with various filters and pagination
- Tests for pipeline name resolution and execution stats inclusion
- Tests for annotation management (set, delete, list)
- Tests for filter parsing and page token handling
- Edge cases like empty results and unsupported filters

### Why make this change?

1. **Refactoring upfront** — structural changes land here so upstream PR diffs only show new functionality, not mixed refactoring noise.
1. **Baseline tests** — establishing coverage now means each upstream PR's test additions map directly to its new features, making review clearer.

### CODE REVIEW

File: cloud_pipelines_backend/api_server_sql.py [hunk 1]
```diff
@@ -165,45 +178,19 @@ def list(
...
-                    filter = filter.replace(
-                        "created_by:me", f"created_by:{current_user}"
-                    )
-                if value:
-                    where_clauses.append(bts.PipelineRun.created_by == value)
-                else:
-                    where_clauses.append(bts.PipelineRun.created_by == None)
-            else:
-                raise NotImplementedError(f"Unsupported filter {filter}.")
+        page_size = 10
```
Comment (Volv-G) [inline cloud_pipelines_backend/api_server_sql.py]: while we're at it, let's define a constant for this maybe?

File: cloud_pipelines_backend/api_server_sql.py [hunk 2]
```diff
@@ -65,6 +71,13 @@ class ListPipelineJobsResponse:
...
 
 
+class ListPipelineRunsParams(BaseModel):
+    filter: str | None = None
+    page_token: str | None = None
+    include_pipeline_names: bool = False
+    include_execution_stats: bool = False
+
+
 import sqlalchemy as sql
```
Comment (Volv-G) [inline cloud_pipelines_backend/api_server_sql.py]: nit: should we move these imports (as well as others) to the top of the file?

File: cloud_pipelines_backend/api_server_sql.py [hunk 3]
```diff
@@ -165,45 +178,19 @@ def list(
         self,
         *,
         session: orm.Session,
-        page_token: str | None = None,
-        # page_size: int  = 10,
-        filter: str | None = None,
         current_user: str | None = None,
-        include_pipeline_names: bool = False,
-        include_execution_stats: bool = False,
+        params: Annotated[ListPipelineRunsParams, Query()],
```
Comment (yuechao-qin) [inline cloud_pipelines_backend/api_server_sql.py]: Revert this back, remove `ListPipelineRunsParams`.

File: cloud_pipelines_backend/api_server_sql.py [hunk 4]
```diff
@@ -4,7 +4,10 @@
 import json
 import logging
 import typing
-from typing import Any, Optional
+from typing import Annotated, Any, Final, Optional
+
+from fastapi import Query
```
Comment (yuechao-qin) [inline cloud_pipelines_backend/api_server_sql.py]: Confirm fastapi is removed from this file, there should be NO fastapi in this module.

File: cloud_pipelines_backend/api_server_sql.py [hunk 5]
```diff
@@ -4,7 +4,10 @@
...
+from pydantic import BaseModel
```
Comment (yuechao-qin) [inline cloud_pipelines_backend/api_server_sql.py]: do not import classes/vars/etc, only modules!

**Review by morgan-wowk — APPROVED** [review]

File: tests/test_api_server_sql.py [hunk 6]
```diff
@@ -55,3 +60,484 @@ def test_accumulate_all_statuses(self):
...
+                    container=structures.ContainerSpec(image="test-image:latest"),
+                ),
+            ),
+        ),
+    )
+
+
+@pytest.fixture()
+def session_factory():
+    engine = sqlalchemy.create_engine(
```
Comment (Ark-kun) [inline tests/test_api_server_sql.py]: Let's use the function from database_ops.

File: cloud_pipelines_backend/api_server_sql.py [hunk 7]
```diff
@@ -26,10 +33,9 @@ def _get_current_time() -> datetime.datetime:
     return datetime.datetime.now(tz=datetime.timezone.utc)
 
 
-from . import component_structures as structures
-from . import backend_types_sql as bts
-from . import errors
-from .errors import ItemNotFoundError
+_PAGE_TOKEN_OFFSET_KEY: Final[str] = "offset"
+_PAGE_TOKEN_FILTER_KEY: Final[str] = "filter"
+_DEFAULT_PAGE_SIZE: Final[int] = 10
```
Comment (Ark-kun) [inline cloud_pipelines_backend/api_server_sql.py]: There is no indication that these constants belong to the PipelineRunsApiService.
If these are only for PipelineRuns then this should be indicate in the name.
Preference: Move these constance to the `PipelineRunsApiService_Sql` class.

File: tests/test_api_server_sql.py [hunk 8]
```diff
@@ -55,3 +60,484 @@ def test_accumulate_all_statuses(self):
...
+
+
+def _create_run(session_factory, service, **kwargs):
+    """Create a pipeline run using a fresh session (mirrors production per-request sessions)."""
+    with session_factory() as session:
+        return service.create(session, **kwargs)
+
+
+class TestPipelineRunServiceList:
+    def test_list_empty(self, session_factory, service):
```
Comment (Ark-kun) [inline tests/test_api_server_sql.py:103]: Let's include simple type annotations in all function signatures.

File: cloud_pipelines_backend/api_server_sql.py [hunk 9]
```diff
@@ -116,7 +118,7 @@ def create(
     def get(self, session: orm.Session, id: bts.IdType) -> PipelineRunResponse:
         pipeline_run = session.get(bts.PipelineRun, id)
         if not pipeline_run:
-            raise ItemNotFoundError(f"Pipeline run {id} not found.")
+            raise errors.ItemNotFoundError(f"Pipeline run {id} not found.")
```
Comment (Ark-kun) [inline cloud_pipelines_backend/api_server_sql.py:119]: Thank you for switching to module imports.
(AFAIR, the reason for this class import was that the error classes were originally declared in this module, but were then extracted into the `errors` module.)

File: tests/test_api_server_sql.py (hunk 8 above)
Comment (yuechao-qin) [reply tests/test_api_server_sql.py:103]: I will add another commit at the top of this stack that does fixing annotations (type hints) for all functions in:
- All test files that I've touched in this stack
- All functions that I've touched in this stack that's not a test file

Reason for adding this as a separate commit is to make restacking issues easier for me.

File: cloud_pipelines_backend/api_server_sql.py [hunk 10]
```diff
@@ -349,6 +337,64 @@ def delete_annotation(
...
+    """
+    where_clauses: list[sql.ColumnElement] = []
+    parsed_filter = _parse_filter(filter_value) if filter_value else {}
+    for key, value in parsed_filter.items():
+        if key == "_text":
+            raise NotImplementedError("Text search is not implemented yet.")
+        elif key == "created_by":
+            if value == "me":
+                if current_user is None:
+                    current_user = ""
```
Comment (morgan-wowk) [inline cloud_pipelines_backend/api_server_sql.py:382]: Question: Shouldn't we raise an error here if an unauthenticated user is trying to filter by `me`? Right now, we end doing `WHERE created_by IS NULL` if there is no `current_user` which does not feel correct
